offchain/types/src/blockchain_config.rs

```diff
@@ -0,0 +1,214 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+use clap::{command, Parser};
+use rollups_events::{Address, Hash};
+use serde::{de::DeserializeOwned, Deserialize};
+use snafu::{ResultExt, Snafu};
+use std::{fs::File, io::BufReader, path::PathBuf};
+
+#[derive(Debug, Snafu)]
+pub enum BlockchainConfigError {
+    #[snafu(display("Json deserialize error"))]
+    JsonDeserializeError { source: serde_json::Error },
+
+    #[snafu(display("Json read error ({})", path.display()))]
+    JsonReadError {
+        path: PathBuf,
+        source: serde_json::Error,
+    },
+
+    #[snafu(display("Read file error ({})", path.display()))]
+    ReadFileError {
+        path: PathBuf,
+        source: std::io::Error,
+    },
+
+    #[snafu(display("Missing configuration: ({})", name))]
+    MissingConfig { name: String },
+}
+
+#[derive(Debug, Parser)]
+#[command(name = "blockchain_config")]
+pub struct BlockchainCLIConfig {
+    /// DApp address
+    #[arg(long, env)]
+    pub dapp_address: Option<String>,
+
+    /// DApp deploy block hash
+    #[arg(long, env)]
+    pub dapp_deploy_block_hash: Option<String>,
+
+    /// History contract address
+    #[arg(long, env)]
+    pub history_address: Option<String>,
+
+    /// Authority contract address
+    #[arg(long, env)]
+    pub authority_address: Option<String>,
+
+    /// Input Box contract address
+    #[arg(long, env)]
+    pub input_box_address: Option<String>,
+
+    /// Path to a file with the deployment json of the dapp
+    #[arg(long, env, default_value = "./dapp_deployment.json")]
+    pub dapp_deployment_file: PathBuf,
+
+    /// Path to file with deployment json of the rollups
+    #[arg(long, env, default_value = "./rollups_deployment.json")]
+    pub rollups_deployment_file: PathBuf,
+}
+
+#[derive(Clone, Debug)]
+pub struct BlockchainConfig {
+    pub dapp_address: Address,
+    pub dapp_deploy_block_hash: Hash,
+    pub history_address: Address,
+    pub authority_address: Address,
+    pub input_box_address: Address,
```

Comment:
So, in the end, we didn't need the deployment hashes of these contracts?
Also, do we need the history address? I think we don't.

Reply:
We only use the hash for the dapp contract. The `Contract` struct is shared.
Going to remove the history contract!

Reply:
Don´t think we can remove the history contract.